### DISCUSSION TITLE
Config rollout

### DISCUSSION THREAD

**Discussion by benjycov** [mKeRix/room-assistant#664] [Q&A]
Hi,

I have rolled out my room-assistant very successfully across my house now, so far 9 nodes (1 running on pi4 alongside HA, and 9 running on PI Zeros) - I have another 2 nodes coming and will potentially add a couple more to cover outdoor spaces like the garage, etc.

Its working really well and now I have my Lovelace interface dynamically updating depending on which room the current user is in, its transformational and very cool.

My problem is that config rollout changes is becoming a bit of a pain (mainly needed for new devices). My thinking is that I can script it and SCP/RSH the configs to each node. The challenge is that each local.yml file is slightly different because of the variables per node (instanceName, peerAddresses and integration specifics such as BLE maxDistance). 

Is there anyway to break up the local.yml into separate files, so for example addresses/overrides can go in a separate include that is common across all nodes?

Is there a better way to do this?

Thanks... Ben

**Comment by mKeRix** [comment]
You can indeed split the config files! The config library that we use supports various [file namings](https://github.com/lorenwest/node-config/wiki/Configuration-Files#file-load-order), which are loaded from the config directory in the order that is specified on the linked page. The config options will be merged except for lists. So if you e.g. want a different set of integrations on one instance you'd have to specify the whole list of integrations again.

Another note that I wanted to make, which may make your life a bit easier: You can use the [ansible setup](https://www.room-assistant.io/guide/quickstart-ansible.html) we provide to manage, upgrade and install room-assistant across many hosts from a single config file. I use it for my own setup as well.

**Reply by benjycov** [reply]
Thanks for the response.

I manned up and spun up ansible,  and attempted to get it to adopt the existing Pi  0 deployments - this led to all sorts of problems and after which I couldn't get the room-assistant service to restart at all on any of the nodes.

So I flattened all of my pi 0's, (now 10 of them) reinstalled Raspian lite from scratch, updated/upgraded them, gave them a hostname, deployed the ssh keys and then let ansible loose do its job.

End result - everything worked perfectly! 

My advice to anyone reading this, if you are moving from a manual install to ansible its easier and cleaner to start from scratch.

11 nodes across a fairly big house (basically 1 node per habitable room) works brilliantly on BLE with the iPhone client. I can wonder around the house and as I enter any room my lovelace interface updates to reflect what that room has to offer (lights/media/etc.).

Thank you so much for this astonishing solution.
